MoltenVK/MoltenVK/Commands/MVKCmdTransfer.mm

```diff
@@ -59,12 +59,6 @@
 
     // Validate
     clearConfigurationResult();
-    if (_srcImage->getMTLPixelFormat() != _dstImage->getMTLPixelFormat()) {
-        setConfigurationResult(mvkNotifyErrorWithText(VK_ERROR_FEATURE_NOT_PRESENT, "vkCmdCopyImage(): The source and destination images must have the same format."));
-    }
-	if ((_srcImage->getMTLTextureType() == MTLTextureType3D) || (_dstImage->getMTLTextureType() == MTLTextureType3D)) {
```

Comment:
Doesn't the `copyFromTexture:...` method take 3D specifications for the origin and size?

Reply:
Hmmm...yes...looking at the Vulkan & Metal docs...I guess copying 3D<->3D should map correctly between Vulkan & Metal...although I haven't tested it.

What won't work...and what triggered me to add that error check...was copying between 3D & 2D images that was introduced with `VK_KHR_maintenance1`.

So I guess the correct error test is to check that either both or neither of the src & dst textures are 3D.

Since you have this PR open...would you mind making that change...keep the existing error test for 3D...but modify it so that it tests for 3D on one side (src | dst) and non-3D on the other?